src/sdk/namespace_fs.js

```diff
@@ -783,13 +793,60 @@ class NamespaceFS {
         await this._load_bucket(params);
         params.mpu_path = this._mpu_path(params);
         try {
-            await fs.promises.stat(params.mpu_path);
+            await nb_native().stat(params.mpu_path);
         } catch (err) {
+            // TOOD: Error handling
             if (err.code === 'ENOENT') err.rpc_code = 'NO_SUCH_UPLOAD';
             throw err;
         }
     }
 
+    async _create_path(dir) {
+        const sep = path.sep;
+        const initDir = path.isAbsolute(dir) ? sep : '';
+        const baseDir = '.';
+
+        return dir.split(sep).reduce(async (parentDir, childDir) => {
```

Comment:
is this code copied from somewhere?
I would not use this without permission to copy.
In any case I don't like how this is written at all.
I would much prefer to write like this:
```js
let dir_path = path.isAbsolute(dir) ? path.sep : '';
for (const item of dir.split(path.sep)) {
    dir_path = path.join(dir_path, item);
    try {
        await nb_native().mkdir(dir_path);
    } catch (err) {
        if (err.code !== 'EEXIST') throw err;
    }
}
```